language_service/src/generate_docs.rs

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#[cfg(test)]
+mod tests;
+
+use crate::display::{increase_header_level, parse_doc_for_summary};
+use crate::{compilation::Compilation, display::CodeDisplay};
+use chrono::Utc;
+use qsc::hir::hir::{Item, ItemKind, Package, Visibility};
+use rustc_hash::FxHashMap;
+use std::fmt::{Display, Formatter, Result};
+use std::fs;
+use std::rc::Rc;
+
+// Warning: this path gets deleted on each run. Use carefully!
+const GENERATED_DOCS_PATH: &str = "../npm/generated_docs";
+
+pub(crate) fn generate_docs(compilation: &Compilation) {
+    let display = &CodeDisplay { compilation };
+
+    delete_existing_docs();
+    fs::create_dir(GENERATED_DOCS_PATH).expect("Unable to create directory for generated docs");
```

Comment:
It's a little odd to be doing file system operations in the language service, being this is usually loaded into a WebWorker in a browser context. I'd expect it to return the content for some other host to do the writing for.

What calls into this? i.e., What is the mechanism by which this is invoked to generate the docs?

Seems maybe we should expose a language service API to get the structure & content of all docs, and then have a Node.js invocation to call into this to fetch the content and write it to disk.

Reply:
File I/O has been moved to the npm layer, per your commits.